frontend/app/(defaultLayout)/products/app-router/[handle]/page.tsx

```diff
@@ -33,5 +42,89 @@ export default async function ProductPage({
 
    if (data.__typename === 'ProductRedirect') redirect(data.target);
 
-   return <div>Product: {data.title}</div>;
+   return (
+      <>
+         <ProductBreadcrumbsJsonLDScript product={data} />
+         <ProductJsonLDScript
+            product={data}
+            selectedVariantId={selectedVariantId(data, searchParams)}
+         />
+         <div>Product: {data.title}</div>
+      </>
+   );
+}
+
+export async function generateMetadata({
+   params,
+   searchParams,
+}: ProductPageProps): Promise<Metadata> {
+   const product = await Product.get(
+      {
+         handle: params.handle,
+         storeCode: DEFAULT_STORE_CODE,
+         ifixitOrigin: devSandboxOrigin() ?? IFIXIT_ORIGIN,
+      },
+      { forceMiss: shouldSkipCache(searchParams) }
+   );
+
+   if (product == null || product.__typename !== 'Product') return {};
+
+   return {
+      title: product.metaTitle,
+      description: product.shortDescription,
+      openGraph: {
+         title: product.metaTitle ?? undefined,
+         description: product.shortDescription ?? undefined,
```

Comment:
I haven't worked with this new metadata api very much, and I'm curious if there is different behavior when these properties are `null` vs `undefined`